### DISCUSSION TITLE
CF2.1 reboots. OW: Cmd 0x22 timeout.

### DISCUSSION THREAD

**Discussion by Noel700** [bitcraze/discussions#844] [Q&A] [answered]
Hi,

currently I'm running my CF2.1 with the multiranger Deck and the Lighthouse Deck. Got the newest Firmware but I tried older Versions as well. At some point in flight (sometimes seconds after takeoff, sometimes a Minute in) the drone decides to reboot. After reboot the LED flashes for self test failed. 
I get this Error out of my Console Log: "OW: Cmd 0x22 timeout."
It also continues with "DECK_INFO: Reading deck nr:0 [FAILED]. No driver will be initialized!
DECK_CORE: 0 deck(s) found"

And at the End: "SYS: The system resumed after a failed assert [WARNING]
SYS: Assert failed at ../src/hal/src/radiolink.c:164
SYS: cfAssertNormalStart [FAIL]"

All the selftests in between finish with "OK".

I found this post as well: https://github.com/orgs/bitcraze/discussions/675
But i don't think that I got a solder Problem. In my Case all the init sensors are OK. 

It happens when I use CFLib as well as when I use the Client to control it. Any Ideas what can be the cause of this Error and how to fix it?

Regards
Noel Egemann

**Comment by krichardsson** [comment]
It looks like there is a problem to read the OneWire memory for some reason. There is a OneWire memory on each deck that holds an identifier that is used to understand which deck(s) that are mounted.

My first guess is that this is a half bad connection to one of the decks or the expansion bus, but it is hard to say. It is also possible that there is a problem with one of the decks, like a bad solder or similar.

An interesting initial test would be to use the system with various combinations of the decks to see if the problem is related to one of the decks. Also wiggle them around a bit to see if you can find any bad connection in the expansion bus. The OW is connected to pin 8 on the right side [of this diagram of the expansion bus](https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#expansion-port-pinout)

I'm not completely sure why you get the assert in the radiolink. There are two CPUs on the Crazyflie: the STM - the main one and the NRF that handles the radio, power and the one wire memory. The communication between the CPUs are done over syslink (including radio and oneWire memory packets), and maybe syslink is blocked when there is a oneWire memory problem? This could explain the assert in the radio link.

I think a good start is to try to figure out if it is related to a deck, the expansion bus or similar.

**Reply by Noel700** [reply]
Hi, thanks for your reply. It looks like there are no bad solders. I measured the connections and they are good from the Chip to the pin. Also the Sensor connections as well as SPI connections are good. Today this problem occured a lot less often. I was able to complete a lot of automatic flights which last about 15-20 seconds each. When manually flying the Drone for testing a few Minutes the Problem didn't occur. Sometimes it happened 30 seconds or so after the Landing and twice during the flight seconds after takeoff. 

So i don't see a real concept behind this, when this happened. From what it looks like it might not be a completely bad solder but maybe a partly bad solder or something completely different. It would be interesting for me what cmd 0x22 is any Ideas about that?

**Reply by tobbeanton** [reply]
My best guess is glitching communication with would originate from bad contact. Could be the connector or some cracked/bad soldering. This only happens at start-up? That is when the 1-wire memory is read (cmd 0x22).

**Reply by Noel700** [reply, marked as answer]
Interestingly another post from me solved my Reboot Problem as well. 4 Lighthouse Basestations are too much for the current Firmware. This caused CPU to run out of resources because the Kalman Estimator got a high priority. Reducing to 2 Basestations solved it for me
